### ISSUE TITLE
OS\Linux wrong distro name

### ISSUE THREAD

**Issue by moharrum** [jrgp/linfo#107]
Hello,

Running Fedora 30 (Thirty) Workstation, I was trying to call `getDistro()`, I got:

```php
array:2 [▼
  "name" => "CentOS"
  "version" => "Fedora release 30 (Thirty)"
]
```

The problem that `Fedora` has `redhat-release` file in `/etc` and looking at `src/Linfo/OS/Linux.php` line 1406 is considered the first match and at line 1471 gets returned as the first correct match.

The last condition is always true since all `$contents_distros` have `distro` index (except ones with `closure`).

```php
            } elseif (isset($distro['distro'])) {
                return array(
                    'name' => $distro['distro'],
                    'version' => $contents,
                );
            }
```

## Edit:

Adding `isset($distro['distro']) && (strpos($distro['distro'], $contents) !== false)` to the last condition seems to solve the issue for me.

```php
        foreach ($contents_distros as $distro) {
            if (!($contents = Common::getContents($distro['file'], false))) {
                continue;
            }

            if (isset($distro['closure']) && ($info = $distro['closure']($contents))) {
                return array(
                    'name' => ucfirst($info['distro']),
                    'version' => $info['version'].(isset($info['codename']) ? ' ('.ucfirst($info['codename']).')' : ''),
                );
            } elseif (isset($distro['regex']) && preg_match($distro['regex'], $contents, $info)) {
                return array(
                    'name' => $distro['distro'],
                    'version' => $info['version'].(isset($info['codename']) ? ' ('.ucfirst($info['codename']).')' : ''),
                );
            } elseif (isset($distro['distro']) && (strpos($distro['distro'], $contents) !== false)) {
                return array(
                    'name' => $distro['distro'],
                    'version' => $contents,
                );
            }
        }
```

**Comment by jrgp** [comment]
Hmm we really should have a unit test for all of these scenarios.

**Comment by jrgp** [comment]
Hopefully c8537c7d1aebdabc873bad9232211fedd6397e09 should fix it. Let me know if it doesn't.

**Comment by jrgp** [comment]
This should be definitely fixed per 22fd93e9440cafa4c49267f0822430dc708ac464
